core/templates/dev/head/pages/exploration_player/ExplorationEngineService.js

```diff
@@ -24,7 +24,7 @@ oppia.constant('INTERACTION_SPECS', GLOBALS.INTERACTION_SPECS);
 // The URL determines which of these it is. Some methods may need to be
 // implemented differently depending on whether the skin is being played
 // in the learner view, or whether it is being previewed in the editor view.
-oppia.factory('ExplorationPlayerService', [
+oppia.factory('ExplorationEngineService', [
```

Comment:
Some thoughts:

- ConversationSkin should call ExplorationEngineService.init(...) and pass in the exploration object. Basically the lifecycle of ExplorationEngineService is that it receives an exploration object to start with, and then just processes incoming commands and requests based on that exploration. Anything else is out of scope for this service.
- Where possible, declare methods on the ExplorationCard object and remove them from here.
- I think the ConversationSkin directive should call StatsReportingService directly when relevant events occur, since ConversationSkin is basically taking care of all the events (navigate to new state, submit answer, etc. -- these are UI things). This helps keep the scope of ExplorationEngineService nice and tight.

Reply:
Done.